sdk/spring-sdk/src/it/java/com/example/wiring/SpringSdkWiringIntegrationTest.java

```diff
@@ -109,6 +108,91 @@ public void verifyCounterEventSourceSubscription() {
             new IsEqual(42 + 1));
   }
 
+  @Test
+  public void verifyActionIsNotSubscribedToMultiplyAndRouterIgnores(){
+
+     webClient
+            .post()
+            .uri("counter/counterId2/increase/1")
+            .retrieve()
+            .bodyToMono(Integer.class)
+            .block(timeout);
+    webClient
+            .post()
+            .uri("counter/counterId2/multiply/2")
+            .retrieve()
+            .bodyToMono(Integer.class)
+            .block(timeout);
+
+    webClient
+            .post()
+            .uri("counter/counterId2/increase/1234")
+            .retrieve()
+            .bodyToMono(Integer.class)
+            .block(timeout);
+
+    Integer counterGet =
+            webClient.get().uri("/counter/counterId2").retrieve().bodyToMono(Integer.class).block(timeout);
+
+    //Before has time to process any action things still work
+    Assertions.assertEquals(1*2 + 1234, counterGet);
+
+    //Once it tries to process actions will fail
+    await()
+            .atMost(10, TimeUnit.SECONDS)
+            .until(
+                    () ->
+                            webClient
+                                    .get()
+                                    .uri("/counter/counterId2")
+                                    .retrieve()
+                                    .bodyToMono(Integer.class)
+                                    .block(timeout),
+                    new IsEqual<Integer>(1*2 + 1234 + 1 ));
```

Comment:
I'm also puzzled by this. How and why this call is now returning the same result +1?

Reply:
added as a comment at the beginning of the assertion